### PR TITLE
Unique peers aliases

### LINKED ISSUE #51: Unique peers aliases
If we talk about cli, it is much easier to work with your peers by their aliases.

Now aliases is not unique, so let's fix that! 

Step 1. Settings should be checked for non-unique aliases on awl starts
Step 2. Add checks on cli peers rename/add commands (serverside)
Step 3. When user add new peer without alias, we should safe new peer's name to alias, to protect users from scam. If new peer's name is not unique we should add something to the end of his name (a number for example)

### PR DESCRIPTION
**Pull request by GrigoryKrasnochub**
Peer`s alias is unique now!
fix #51

### CODE REVIEW

File: application_test.go [hunk 1]
```diff
@@ -172,6 +172,28 @@ func TestAutoAcceptFriendRequest(t *testing.T) {
...
+	defer peer3.Close()
+	ensurePeersAvailableInDHT(a, peer1, peer3)
+
+	err := peer1.api.SendFriendRequest(peer2.PeerID(), "peer")
+	a.NoError(err)
+
+	time.Sleep(200 * time.Millisecond)
+
+	err = peer1.api.SendFriendRequest(peer3.PeerID(), "peer")
+	a.Error(err)
```
Comment (pymq) [inline application_test.go:194]: I think we should also check what kind of error we are getting here, checking for substring should be enough

File: config/config.go [hunk 2]
```diff
@@ -97,6 +100,24 @@ func (c *Config) Save() {
 	c.RUnlock()
 }
 
+func (c *Config) IsUniqPeerAlias(alias string) bool {
+	c.Lock()
```
Comment (pymq) [inline config/config.go:104]: Why not RLock?

File: config/config.go [hunk 3]
```diff
@@ -97,6 +100,24 @@ func (c *Config) Save() {
...
+	for _, kPeer := range c.KnownPeers {
+		if kPeer.Alias == alias {
+			return false
+		}
+	}
+	return true
+}
+
+func (c *Config) GenUniqPeerAlias(name, alias string) string {
+	c.Lock()
```
Comment (pymq) [inline config/config.go:115]: Why not RLock?

File: config/other.go [hunk 4]
```diff
@@ -169,11 +169,17 @@ func setDefaults(conf *Config, bus awlevent.Bus) {
...
 
+	uniqAliases := make(map[string]struct{}, len(conf.KnownPeers))
 	if conf.KnownPeers == nil {
 		conf.KnownPeers = make(map[string]KnownPeer)
 	}
 	for peerID := range conf.KnownPeers {
 		peer := conf.KnownPeers[peerID]
+		newAlias := conf.genUniqPeerAlias(peer.Name, peer.Alias, uniqAliases)
+		if newAlias != peer.Alias {
+			logger.Warnf("incorrect config: peer alias %s is not unique, updated automaticaly to %s", peer.Alias, newAlias)
```
Comment (pymq) [inline config/other.go:180]: I think we should also print the peerID here to simplify debugging, to know which of the same peers was renamed to which name

**Review by pymq — APPROVED** [review]

File: config/config.go (hunk 2 above)
Comment (GrigoryKrasnochub) [reply config/config.go:104]: I forget to change it when moved on to no-cache solution. Thanks!

**Review by pymq — APPROVED** [review]

**Pull request merged**